plugins/OnlineSales.Plugin.SendGrid/Controllers/SendgridController.cs

```diff
@@ -0,0 +1,281 @@
+﻿// <copyright file="SendgridController.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.Collections.Immutable;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using OnlineSales.Data;
+using OnlineSales.Entities;
+using OnlineSales.Interfaces;
+using OnlineSales.Plugin.SendGrid.Data;
+using OnlineSales.Plugin.SendGrid.DTOs;
+using OnlineSales.Plugin.SendGrid.Entities;
+using OnlineSales.Plugin.SendGrid.Exceptions;
+using SendGrid.Helpers.EventWebhook;
+using Serilog;
+
+namespace OnlineSales.Plugin.SendGrid.Controllers;
+
+[Route("api/[controller]")]
+public class SendgridController : ControllerBase
+{
+    private static readonly int BatchSize = 10;
```

Comment:
@IgorKostyukov 10 records is a very small batch size. I think it can be 500 or even 1000 records easily.